source/common/config/utility.h

```diff
@@ -241,6 +241,17 @@ class Utility {
                                    POOL_HISTOGRAM(scope))};
   }
 
+
+
+  /**
+   * Get the hash map for this factory.
+   */
+  template <class Factory>
+  static absl::flat_hash_map<std::string, Factory*>&  getFactoryMap() {
```

Comment:
I don't like exposing the representation in the API; can we instead use a forEach model, where you call a std::function on every element?